### PR TITLE
(Fixes #52) Support cancel workflow

### LINKED ISSUE #52: Allow terminate a running workflow which would gracefully shut down all resources

### PR DESCRIPTION
**Pull request by realstraw**

### CODE REVIEW

**Review by damianxu88 — APPROVED** [review]
LGTM

File: orchard-core/src/main/scala/com/salesforce/mce/orchard/system/OrchardSystem.scala [hunk 1]
```diff
@@ -7,34 +7,62 @@
...
+        ctx.watchWith(workflowMgr, WorkflowTerminated(workflowId))
+        apply(ctx, database, timers, workflows + (workflowId -> workflowMgr))
+      }
+
+    case WorkflowTerminated(workflowId) =>
+      ctx.log.info(s"${ctx.self} Received WorkflowTerminated($workflowId)")
+      apply(ctx, database, timers, workflows - workflowId)
+
+    case ScanCanceling =>
+      ctx.log.info(s"${ctx.self} Received ScanCencling")
```
Comment (schowsf) [inline orchard-core/src/main/scala/com/salesforce/mce/orchard/system/OrchardSystem.scala:58]: typo. ScanCanceling

**Review by schowsf — APPROVED** [review]
LGTM

File: orchard-core/src/main/scala/com/salesforce/mce/orchard/system/OrchardSystem.scala [hunk 2]
```diff
@@ -7,34 +7,62 @@
...
 
-import com.salesforce.mce.orchard.db.OrchardDatabase
+import com.salesforce.mce.orchard.db.{OrchardDatabase, WorkflowQuery}
+import com.salesforce.mce.orchard.model.Status
 import com.salesforce.mce.orchard.system.actor.WorkflowMgr
+import akka.actor.typed.scaladsl.TimerScheduler
 
 object OrchardSystem {
 
+  val CancelingScanDelay = 10.seconds
```
Comment (anniezhang77) [inline orchard-core/src/main/scala/com/salesforce/mce/orchard/system/OrchardSystem.scala:22]: 10 sec?

**Pull request merged**